docs/setting-up/client/mongodb.md

```diff
@@ -45,128 +62,198 @@ db.getSiblingDB("admin").createUser({
+**Example -- connect via UNIX socket**
 
 ```sh
-pmm-admin add mongodb --username=pmm --password=pmm --service-name=mongo --host=127.0.0.1 --port=27017
+pmm-admin add mongodb --socket=/tmp/mongodb-27017.sock
 ```
 
-You can add a MongoDB instance using a UNIX socket with the `--socket` option:
+
+**Example -- connecting via SSL/TLS**
 
 ```sh
-pmm-admin add mongodb --socket=/tmp/mongodb-27017.sock
+pmm-admin add mongodb --tls \
+--tls-certificate-key-file=PATHTOCER \
+--tls-certificate-key-file-password=IFPASSWORDTOCERTISSET \
+--tls-ca-file=PATHTOCACERT
 ```
 
-> If the password contains special symbols like the 'at' (`@`) symbol, the host might not be detected correctly. Make sure that you insert the password with special characters replaced with their escape sequences. The simplest way is to use the [`encodeURIComponent`][ENCODE_URI] JavaScript function in your browser's web console (usually found under *Development Tools*). Evaluate the function with your password as the parameter. For example:
+where:
+
+- `PATHTOCERT`: Path to TLS certificate file.
+- `IFPASSWORDTOCERTISSET`: Password for TLS certificate file.
+- `PATHTOCACERT`: Path to certificate authority file.
+
+> **See also**
+>
+> - [`pmm-admin` man page for `pmm-admin add mongodb`](../../details/commands/pmm-admin.md#mongodb)
 >
-> ```javascript
-> encodeURIComponent('$ecRet_pas$w@rd')
-> "%24ecRet_pas%24w%40rd"
-> ```
+> - [Troubleshooting connection difficulties][TROUBLESHOOTING_CONNECTION]
+
+## Tips
+
+- When adding nodes of a replica set, add each node separately using the same cluster name.
 
-[ENCODE_URI]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent
 
-## Passing SSL parameters to the MongoDB monitoring service
+## Check the service
 
-SSL/TLS related parameters are passed to an SSL enabled MongoDB server as
-monitoring service parameters along with the `pmm-admin add` command when adding
-the MongoDB monitoring service.
+**Check service - PMM user interface**
 
-Run this command as root or by using the `sudo` command
+1. Select *{{icon.cog}} Configuration-->{{icon.inventory}} PMM Inventory-->{{icon.inventory}} Inventory list*.
+2. Look in the *Services* tab for a matching *Service Type* (MongoDB), *Service name*, *Addresses*, and any other values used when adding the service.
+3. Look in the *Agents* tab to check the desired data source is being used.
+
+**Check service - Command line**
+
+Look for your service in the output of this command.
 
 ```sh
-pmm-admin add mongodb --tls
+pmm-admin inventory list services --service-type=mongodb
 ```
 
-**Supported SSL/TLS Parameters**
+**Check data**
+
+1. Open the *MongoDB Instances Overview* dashboard.
+2. Set the *Service Name* to the newly-added service.
```

Comment:
I have a question, putting myself in the shoes of a normal MongoDB user: Why do I have to set a service name? MongoDB is a mongod (or mongos, which can be detected). Each node has a unique identity - hostname and port - and I'm not accustomed to caring about identity of the node other than what replica set it is of (which can also be automatically detected, without requiring any authorization either).

Reply:
@rnovikovP ?

Reply:
"Autodetection for labels"  (and "service-name" is one of the labels) not yet implemented in PMM.. so we need to ask them for now

Reply:
Ack.